### ISSUE TITLE
Suppress need for `require` in `require-returns` when `@implements` present  

### ISSUE THREAD

**Issue by brettz9** [gajus/eslint-plugin-jsdoc#201]
While normally the issue of a function serving as a class may be handled by #200 , the fact that http://usejsdoc.org/tags-implements.html does not exclude the possibility of a normal function using `@implements` to point to a function signature that it implements, I would think this case should be suppressed as well. @sandersn

**Comment by brettz9** [comment]
And if instead, TypeScript disallows this, one may wish to instead *prevent* `@implements` on a function unless that function is marked with `@class`.

**Comment by sandersn** [comment]
+1 for adding the option to prevent `@implements` on functions. This is exactly the kind of jsdoc error we can't put in Typescript itself because we don't want to be noisy, but it's perfect for eslint.

**Comment by brettz9** [comment]
Ok, I agree that is a good option, but until jsdoc may implement `typeof` (which IIUC, is the correct replacement in Typescript; see also https://github.com/jsdoc3/jsdoc/issues/1349 and https://github.com/jsdoctypeparser/jsdoctypeparser/issues/50 ), I think jsdoc users should have the ability to suppress the error.

**Comment by brettz9** [comment]
Also, does Typescript have any way to easily indicate a signature of an options object which has say several properties and callbacks?

`typeof` seems to assume that there is already an implementation in code, no?

`@typedef` might work, e.g., with a type of "function" for callback properties, but this wouldn't allow a full function signature.

And `@interface` seems it might be an approach to use, as, per http://usejsdoc.org/tags-interface.html#virtual-comments , but I'm not sure on whether it is intended to be used for the likes of object literals.

**Comment by sandersn** [comment]
Is this what you mean for "indicate a signature of an options object which has several properties and callbacks"?

```js
/** @typedef {{
  (n: number): number
  p: number
  opt?: number
  m: (n: number) => number
}} Foo */

`Foo` is callable, has a method `m` and two other properties `p` and `opt`. `opt` is optional. You can use `@callback` if you want to provide documentation for `m`'s signature separately. I don't think it's possible for the signature of `Foo` itself.

**Comment by brettz9** [comment]
Ah, ok... Very nice, thanks...

**Comment by brettz9** [comment]
Fixed by #217.
